### PR TITLE
Add getTrustlines and unit test

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: src/api/ledger/trustlines.js [hunk 1]
```diff
@@ -1,127 +1,48 @@
...
-      trustlines.trustlines = result.lines;
-
-      resolve(callback(null, trustlines));
-    });
+  });
+}
 
-    return promise;
-  }
+/*:: type Options = {currency: string, counterparty: string} */
```
Comment (clark800) [inline src/api/ledger/trustlines.js]: Add `limit` and `ledgerVersion`

File: src/api/ledger/parse/account-trustline.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+const utils = require('./utils');
+
+// rippled 'account_lines' returns a different format for
+// trustlines than 'tx'
+function parseAccountTrustline(address, trustline) {
```
Comment (clark800) [inline src/api/ledger/parse/account-trustline.js]: `address` is unused

File: src/api/ledger/utils.js [hunk 3]
```diff
@@ -6,6 +6,29 @@ const asyncify = require('simple-asyncify');
...
+// If the marker is omitted from a response, you have reached the end
+// getter(marker, limit, callback), callback(error, {marker, results})
+function getRecursiveRecur(getter, marker, limit, callback) {
+  getter(marker, limit, (error, data) => {
+    if (error) {
+      return callback(error);
+    }
+    const remaining = limit - data.results.length;
+    if (remaining > 0 && data.marker !== undefined) {
+      getRecursiveRecur(getter, data.marker, remaining, (_error, _data) => {
```
Comment (clark800) [inline src/api/ledger/utils.js]: `_data` -> `results`

**Comment by hserang** [comment]
LGTM! Thank you for walking me through the PR.

**Pull request merged**